### PR TITLE
Fix duplicating / multiplicating support blockers

### PR DESCRIPTION
**Pull request by nallath**
CURA-7851

### CODE REVIEW

**Comment by nallath** [comment]
This should provide a fix for  https://github.com/Ultimaker/Cura/issues/8776

File: cura/Arranging/Nest2DArrange.py [hunk 1]
```diff
@@ -110,18 +110,11 @@ def findNodePlacement(nodes_to_arrange: List["SceneNode"], build_volume: "BuildV
...
-    :param factor: The library that we use is int based. This factor defines how accuracte we want it to be.
-    :param add_new_nodes_in_scene: Whether to create new scene nodes before applying the transformations and rotations
-
-    :return: found_solution_for_all: Whether the algorithm found a place on the buildplate for all the objects
-    """
+def createGroupOperationForArrange(nodes_to_arrange: List["SceneNode"],
+                                   build_volume: "BuildVolume",
+                                   fixed_nodes: Optional[List["SceneNode"]] = None,
+                                   factor = 10000,
+                                   add_new_nodes_in_scene: bool = False):
```
Comment (konskarm) [inline cura/Arranging/Nest2DArrange.py:117]:
```suggestion
                                   add_new_nodes_in_scene: bool = False)  -> Tuple[GroupedOperation, int]:
```

File: cura/Arranging/Nest2DArrange.py [hunk 2]
```diff
@@ -141,8 +134,30 @@ def arrange(nodes_to_arrange: List["SceneNode"], build_volume: "BuildVolume", fi
         else:
             # We didn't find a spot
             grouped_operation.addOperation(
-                TranslateOperation(node, Vector(200, node.getWorldPosition().y, -not_fit_count * 20), set_position = True))
+                TranslateOperation(node, Vector(200, node.getWorldPosition().y, -not_fit_count * 20),
+                                   set_position=True))
```
Comment (konskarm) [inline cura/Arranging/Nest2DArrange.py:138]:
```suggestion
                TranslateOperation(node, Vector(200, node.getWorldPosition().y, -not_fit_count * 20), set_position = True))
```

File: cura/MultiplyObjectsJob.py [hunk 3]
```diff
@@ -56,19 +58,33 @@ def run(self) -> None:
...
         found_solution_for_all = True
         if nodes:
-            found_solution_for_all = arrange(nodes, Application.getInstance().getBuildVolume(), fixed_nodes,
-                                             factor = 10000, add_new_nodes_in_scene = True)
+            group_operation, not_fit_count = createGroupOperationForArrange(nodes,
+                                                                            Application.getInstance().getBuildVolume(),
+                                                                            fixed_nodes,
+                                                                            factor=10000, add_new_nodes_in_scene=True)
+        else:
+            group_operation = GroupedOperation()
```
Comment (konskarm) [inline cura/MultiplyObjectsJob.py:81]: This whole if statement can be simplified as follows:

``` python
group_operation = GroupedOperation()
if nodes:
    group_operation, not_fit_count = createGroupOperationForArrange(nodes,
                                                                            Application.getInstance().getBuildVolume(),
                                                                            fixed_nodes,
                                                                            factor = 10000, add_new_nodes_in_scene = True)
```

File: cura/MultiplyObjectsJob.py [hunk 4]
```diff
@@ -56,19 +58,33 @@ def run(self) -> None:
...
-                nodes.append(new_node)
+                if not current_node.getParent().callDecoration("isSliceable"):
+                    nodes.append(new_node)
+                else:
+                    # The node we're trying to place has another node that is sliceable as a parent.
+                    # As such, we shouldn't arrange it (but it should be added to the scene!)
+                    nodes_to_add_without_arrange.append(new_node)
+                    new_node.setParent(current_node.getParent())
 
         found_solution_for_all = True
```
Comment (konskarm) [inline cura/MultiplyObjectsJob.py:74]: This variable seems to always be True now. Shouldn't it change based on the outcome of the `createGroupedOperationForArrange()`? 
If it doesn't, then the message `no_full_solution_message` later on can be removed.

**Review by konskarm — CHANGES_REQUESTED** [review]
Some code-style changes and a request to simplify a small part. 
Also, bit of a nitpick, but I think that the new eraser node that is added should be moved at least a tiny bit. Right now it's not visible that it is multiplied if you don't have the Object list in the bottom left corner expanded, because it the new Eraser block is right under the old one.

**Pull request merged**